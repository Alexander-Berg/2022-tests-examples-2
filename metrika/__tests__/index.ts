import { remap } from '../normalize';

describe('remap', () => {
    it('works', () => {
        const result = remap(
            { yandexEmployeeLogin: '13' },
            { 13: 'loginFromStaff' },
        );

        expect(result).toEqual({ yandexEmployeeLogin: 'loginFromStaff' });
    });

    it('skips keys if they not presented in raw object', () => {
        const result = remap({ yandexEmployeeLogin: '13' }, {});

        expect(result).toEqual({});
    });

    it('skips keys if they not presented in mapping object', () => {
        const result = remap({}, { 13: 'loginFromStaff' });

        expect(result).toEqual({});
    });

    it('mixed case', () => {
        const result = remap(
            { someOtherField: '42' },
            { 13: 'loginFromStaff' },
        );

        expect(result).toEqual({});
    });
});
