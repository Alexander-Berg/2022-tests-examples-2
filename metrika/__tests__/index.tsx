import * as React from 'react';

import { render, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import { Highlight, Substr } from '..';
configure({ adapter: new Adapter() });

describe('Highlight', () => {
    it('работает по строке', () => {
        const container = render(
            <div>
                <Highlight pattern="екс">текст</Highlight>
            </div>,
        );

        expect(container).toMatchSnapshot();
    });
    it('работает с вложенными строками', () => {
        const container = render(
            <div>
                <Highlight pattern="екс">
                    <span>текст</span>
                </Highlight>
            </div>,
        );

        expect(container).toMatchSnapshot();
    });
    it('работает с несколькими вложенными строками', () => {
        const container = render(
            <div>
                <Highlight pattern="екс">
                    <span>текст</span>
                    <span>текст</span>
                </Highlight>
            </div>,
        );

        expect(container).toMatchSnapshot();
    });
    it('работает с несколькими паттернами через пробел', () => {
        const container = render(
            <div>
                <Highlight pattern="ако екс">
                    <span>какой-то текст</span>
                    <span>какой-то текст</span>
                </Highlight>
            </div>,
        );

        expect(container).toMatchSnapshot();
    });
    it('работает с паттернами через массив', () => {
        const container = render(
            <div>
                <Highlight pattern={['ако', 'екс']}>
                    <span>какой-то текст</span>
                    <span>какой-то текст</span>
                </Highlight>
            </div>,
        );

        expect(container).toMatchSnapshot();
    });
    it('работает с подсветкой фона', () => {
        const container = render(
            <div>
                <Highlight pattern={['ако', 'екс']} highlight="background">
                    <span>какой-то текст</span>
                    <span>какой-то текст</span>
                </Highlight>
            </div>,
        );

        expect(container).toMatchSnapshot();
    });
    it('работает позиционная подсветка', () => {
        const container = render(
            <div>
                <Substr match={{ start: 1, end: 3 }}>Here comes Johny!</Substr>
            </div>,
        );

        expect(container).toMatchSnapshot();
    });
});
