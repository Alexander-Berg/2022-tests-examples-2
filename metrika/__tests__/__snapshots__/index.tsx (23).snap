// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`direction asc 1`] = `
.c2 {
  position: absolute;
  top: 4px;
  right: -14px;
  -webkit-transform: rotate(180deg);
  -ms-transform: rotate(180deg);
  transform: rotate(180deg);
}

.c0 {
  position: relative;
  display: inline-block;
  font-weight: bold;
  color: #7f8285;
  text-align: right;
  vertical-align: top;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  line-height: 17px;
  cursor: pointer;
  color: #000000;
}

.c0:hover {
  color: #000000;
}

.c1 {
  white-space: normal;
  vertical-align: top;
}

<span
  class="c0"
>
  <span
    class="c1"
  >
    Заголовок
  </span>
  <span
    class="BaseIcon IconSortTriangle c2"
    data-lego="react"
  >
    <svg
      fill="currentColor"
      height="12px"
      viewBox="0 0 12 12"
      width="12px"
    >
      <path
        d="M3 3.032h6L6 8.97z"
      />
    </svg>
  </span>
</span>
`;

exports[`direction desc 1`] = `
.c2 {
  position: absolute;
  top: 4px;
  right: -14px;
}

.c0 {
  position: relative;
  display: inline-block;
  font-weight: bold;
  color: #7f8285;
  text-align: right;
  vertical-align: top;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  line-height: 17px;
  cursor: pointer;
  color: #000000;
}

.c0:hover {
  color: #000000;
}

.c1 {
  white-space: normal;
  vertical-align: top;
}

<span
  class="c0"
>
  <span
    class="c1"
  >
    Заголовок
  </span>
  <span
    class="BaseIcon IconSortTriangle c2"
    data-lego="react"
  >
    <svg
      fill="currentColor"
      height="12px"
      viewBox="0 0 12 12"
      width="12px"
    >
      <path
        d="M3 3.032h6L6 8.97z"
      />
    </svg>
  </span>
</span>
`;

exports[`with default params 1`] = `
.c0 {
  position: relative;
  display: inline-block;
  font-weight: bold;
  color: #7f8285;
  text-align: right;
  vertical-align: top;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  line-height: 17px;
  cursor: pointer;
}

.c0:hover {
  color: #000000;
}

.c1 {
  white-space: normal;
  vertical-align: top;
}

<span
  class="c0"
>
  <span
    class="c1"
  >
    Заголовок
  </span>
</span>
`;

exports[`with direction and hint 1`] = `
.c2 {
  position: absolute;
  top: 4px;
  right: -14px;
  -webkit-transform: rotate(180deg);
  -ms-transform: rotate(180deg);
  transform: rotate(180deg);
}

.c0 {
  position: relative;
  display: inline-block;
  font-weight: bold;
  color: #7f8285;
  text-align: right;
  vertical-align: top;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  line-height: 17px;
  cursor: pointer;
  color: #000000;
}

.c0:hover {
  color: #000000;
}

.c1 {
  white-space: normal;
  vertical-align: top;
}

<span
  class="c0"
>
  <span
    class="c1"
  >
    Заголовок
  </span>
  <span
    class="BaseIcon IconSortTriangle c2"
    data-lego="react"
  >
    <svg
      fill="currentColor"
      height="12px"
      viewBox="0 0 12 12"
      width="12px"
    >
      <path
        d="M3 3.032h6L6 8.97z"
      />
    </svg>
  </span>
</span>
`;
